src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadmePreviewViewModel.cs

```diff
@@ -18,23 +18,41 @@ public sealed class ReadmePreviewViewModel : TitledPageViewModelBase
         private string _rawReadme;
         private DetailedPackageMetadata _packageMetadata;
         private bool _canRenderLocalReadme;
+        private bool _isBusy;
 
         public ReadmePreviewViewModel(INuGetPackageFileService packageFileService, ItemFilter itemFilter, bool isReadmeFeatureEnabled)
         {
             _nugetPackageFileService = packageFileService ?? throw new ArgumentNullException(nameof(packageFileService));
             _canRenderLocalReadme = CanRenderLocalReadme(itemFilter);
             _nugetPackageFileService = packageFileService;
             _errorLoadingReadme = false;
+            _isBusy = false;
             _rawReadme = string.Empty;
             _packageMetadata = null;
             Title = Resources.Label_Readme_Tab;
             IsVisible = isReadmeFeatureEnabled;
         }
 
+        public bool IsReadmeReady { get => !IsBusy && !ErrorLoadingReadme; }
+
         public bool ErrorLoadingReadme
```

Comment:
ErrorLoadingReadme should not be public and called from an outside component. The View should not be setting a state property in the ViewModel when a UI component fails to set the content correctly. This violates the responsibilities of the View and ViewModel in MVVM.

Loading the Readme and Rendering the readme are two separate concepts. The ViewModel loads the Readme, the View renders it. The initial concern is with the name of ErrorLoadingReadme in this context. This property is actually dealing with an error when the control fails to update the markdown due to an argument exception or invalid operation, not when the readme fails to load. The second concern is how the View then sets a property on the ViewModel which then is bound back to the View, for an error that occurred on the View side.

Reply:
For now I'm renaming ErrorLoadingReadme to ErrorWithReadme, so that the name better reflects it's usage. I'll look into what it would take to handle the errors properly so that the View isn't updating the ViewModel when there's an error trying to render the README.